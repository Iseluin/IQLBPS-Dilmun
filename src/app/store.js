
import {
  configureStore
} from '@reduxjs/toolkit';
import userReducer from '../features/user/userSlice';
import itemReducer from '../features/slices/item'

export const store = configureStore({
  reducer: {
    user: userReducer,
    item: itemReducer
  },
});
