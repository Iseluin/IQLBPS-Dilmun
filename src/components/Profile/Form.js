// import { first } from "lodash";
import React, { useState } from 'react';

import { useForm } from 'react-hook-form';
import { useTranslation } from 'react-i18next';

function Form({ toggleForm }) {
  const { t } = useTranslation();
  const {
    register,
    // handleSubmit,
    //  reset
  } = useForm();

  const { firstName, setfirstName } = useState('');
  const { lastName, setlastName } = useState('');
  const { country, setcountry } = useState('');
  const { city, setcity } = useState('');
  const { error, setError } = useState('');

  const {
    submitForm,
  } = (e) => {
    e.preventDefault();
    if (
      firstName.length === 0 ||
      lastName.length === 0 ||
      country.length === 0 ||
      city.length === 0
    ) {
      setError(true);
    }
  };
  // const closeModal = React.useCallback(() => setIsOpen(false) , [])

  return (
    <form className="flex flex-col justify-center items-center content-center w-5/6 bg-primary rounded-lg ">
      <div className="bg- py-6 px-10 sm:max-w-md w-full rounded-md">
        <div className="sm:text-3xl text-2xl font-semibold text-center w-full text-primary  mb-12">
          {t('profile.edit_your_profile')}
        </div>
        {/* Image upload component */}
        <div className="flex justify-center mt-6 ">
          <div className="w-full rounded-lg shadow-xl bg-background border-dashed hover:border-primary">
            <div className="m-4">
              <p className=" mb-3 text-primary">
                {t('profile.profile_picture_upload')}
              </p>
              <div className="flex items-center justify-center w-full">
                <div className="flex flex-col border-4 border-dashed hover:border-primary">
                  <div className="flex flex-col items-center justify-center pt-5">
                    <svg
                      xmlns="http://www.w3.org/2000/svg"
                      className="w-full h-6 text-primary group-hover:text-background"
                      fill="none"
                      viewBox="0 0 24 24"
                      stroke="currentColor"
                    >
                      <path d="M7 16a4 4 0 01-.88-7.903A5 5 0 1115.9 6L16 6a5 5 0 011 9.9M15 13l-3-3m0 0l-3 3m3-3v12" />
                    </svg>
                    <p className="pt-1 text-sm tracking-wider text-primary">
                      {t('profile.attach_a_file')}
                    </p>
                  </div>
                  <input
                    type="file"
                    className="opacity-0"
                    {...register('profilePic')}
                  />
                </div>
              </div>
              {/* Background Image */}
              <div className="m-4">
                <p className="inline-block mb-3 text-primary">
                  {t('profile.background_picture_upload')}
                </p>
                <div className="flex items-center justify-center w-full">
                  <div className="flex flex-col  border-4 border-dashed hover:border-primary">
                    <div className="flex flex-col items-center justify-center pt-5">
                      <svg
                        xmlns="http://www.w3.org/2000/svg"
                        className="w-6 h-6 text-primary group-hover:text-background"
                        fill="none"
                        viewBox="0 0 24 24"
                        stroke="currentColor"
                      >
                        <path d="M7 16a4 4 0 01-.88-7.903A5 5 0 1115.9 6L16 6a5 5 0 011 9.9M15 13l-3-3m0 0l-3 3m3-3v12" />
                      </svg>
                      <p className="pt-1 text-sm tracking-wider text-primary">
                        {t('profile.attach_a_file')}
                      </p>
                    </div>
                    <input
                      type="file"
                      className="opacity-0"
                      {...register('backgroundPic')}
                    />
                  </div>
                </div>
              </div>
            </div>
            <div className="flex justify-center p-2">
              <button
                type="button"
                className=" sm:w-96 w-80 font-semibold shadow-lg my-2 bg-primary text-background py-2 rounded-md hover:bg-tertiary hover:text-primary duration-500 "
              >
                {t('profile.upload')}
              </button>
            </div>
          </div>
        </div>
        <div className="flex flex-col justify-center gap-2 items-center">
          <input
            type="text"
            onChange={(e) => setfirstName(e.target.value)}
            className=" w-full p-1 bg-background rounded-lg mt-3 "
            placeholder={t('profile.first_name')}
            {...register('firstName', { required: true, maxLength: 10 })}
          />
          {error ? <p>{t('error.first_name')}</p> : ''}

          <input
            type="text"
            onChange={(e) => setlastName(e.target.value)}
            className=" w-full p-1 bg-background rounded-lg "
            placeholder={t('profile.last_name')}
            {...register('lastName')}
          />
          {error ? <p>{t('error.last_name')}</p> : ''}
          <input
            type="text"
            onChange={(e) => setcountry(e.target.value)}
            className=" w-full p-1 bg-background rounded-lg"
            placeholder={t('profile.country')}
            {...register('country')}
          />
          {error ? <p>{t('error.country')}</p> : ''}
          <input
            type="text"
            onChange={(e) => setcity(e.target.value)}
            className=" w-full p-1 bg-background rounded-lg"
            placeholder={t('profile.city')}
            {...register('city')}
          />
          {error ? <p>{t('error.city')}</p> : ''}
          <button
            type="button"
            onClick={submitForm}
            className="sm:w-96 w-80 font-semibold shadow-lg py-2 bg-primary text-background rounded-md hover:bg-tertiary hover:text-primary duration-500"
          >
            {t('profile.save')}
          </button>
          <button
            type="button"
            onClick={toggleForm}
            className=" sm:w-96 w-80 font-semibold shadow-lg py-2 bg-primary text-background  rounded-md hover:bg-tertiary hover:text-primary duration-500"
          >
            {t('profile.cancel')}
          </button>
        </div>
      </div>
    </form>
  );
}
export default Form;
