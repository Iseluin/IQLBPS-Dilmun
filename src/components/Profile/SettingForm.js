import { useForm } from 'react-hook-form';
// import { useNavigate } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { updateUserInfo, user } from '../../features/slices/user';
// import { PROFILE } from '../../route';

function SettingForm({ toggleForm }) {
  const userData = useSelector(user);
  const dispatch = useDispatch();
  // const navigate = useNavigate();

  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm({
    defaultValues: {
      firstName: userData.firstName ? userData.firstName : '',
      lastName: userData.lastName ? userData.lastName : '',
      city: userData.address.city ? userData.address.city : '',
      country: userData.address.country ? userData.address.country : '',
    },
  });

  const onSubmit = (data) => {
    const { firstName, lastName, city, country } = data;
    dispatch(
      updateUserInfo({
        user: userData,
        firstName,
        lastName,
        city,
        country,
      })
    );
    toggleForm();
  };

  return (
    <div className="justify-center items-center flex overflow-x-hidden overflow-y-auto fixed inset-0 z-50 outline-none focus:outline-none">
      <div className="relative w-auto my-6 mx-auto max-w-3xl">
        <div className="border-0 rounded-lg shadow-lg relative flex flex-col w-full bg-white outline-none focus:outline-none">
          <div className="flex items-start justify-between p-5 border-b border-solid border-slate-200 rounded-t">
            <h3 className="text-3xl text-primary font-semibold">
              Edit your profile
            </h3>
          </div>
          {/* <div className="relative p-6 flex-auto"> */}
          <form
            className="relative p-6 flex-auto"
            onSubmit={handleSubmit(onSubmit)}
          >
            <div className="  px-10 sm:max-w-md w-full rounded-md">
              <div className="flex justify-center mt-6 ">
                <div className="w-full rounded-lg  bg-background border-dashed hover:border-primary">
                  <div className="m-4">
                    <p className=" mb-3 text-primary">Profile picture upload</p>
                    <div className="flex items-center justify-center w-full">
                      <div className="flex flex-col border-4 border-dashed hover:border-primary">
                        <div className="flex flex-col items-center justify-center pt-5">
                          <svg
                            xmlns="http://www.w3.org/2000/svg"
                            className="w-full h-6 text-primary group-hover:text-background"
                            fill="none"
                            viewBox="0 0 24 24"
                            stroke="currentColor"
                          >
                            <path d="M7 16a4 4 0 01-.88-7.903A5 5 0 1115.9 6L16 6a5 5 0 011 9.9M15 13l-3-3m0 0l-3 3m3-3v12" />
                          </svg>
                          <p className="pt-1 text-sm tracking-wider text-primary">
                            Attach a file
                          </p>
                        </div>
                        <input
                          type="file"
                          className="opacity-0"
                          {...register('profilePic')}
                        />
                      </div>
                    </div>
                    {/* Background Image */}
                    <div className="m-4">
                      <p className="inline-block mb-3 text-primary">
                        Background picture upload
                      </p>
                      <div className="flex items-center justify-center w-full">
                        <div className="flex flex-col  border-4 border-dashed hover:border-primary">
                          <div className="flex flex-col items-center justify-center pt-5">
                            <svg
                              xmlns="http://www.w3.org/2000/svg"
                              className="w-6 h-6 text-primary group-hover:text-background"
                              fill="none"
                              viewBox="0 0 24 24"
                              stroke="currentColor"
                            >
                              <path d="M7 16a4 4 0 01-.88-7.903A5 5 0 1115.9 6L16 6a5 5 0 011 9.9M15 13l-3-3m0 0l-3 3m3-3v12" />
                            </svg>
                            <p className="pt-1 text-sm tracking-wider text-primary">
                              Attach a file
                            </p>
                          </div>
                          <input
                            type="file"
                            className="opacity-0"
                            {...register('backgroundPic')}
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div className="flex pb-6 flex-col justify-center gap-2 items-center">
                <input
                  type="text"
                  className=" w-full p-1 bg-background rounded-lg mt-3 "
                  placeholder=" First name "
                  {...register('firstName', {
                    required: true,
                    maxLength: 10,
                  })}
                />
                {errors.firstName ? (
                  <p className="text-red-800 font-semibold text-[12px]">
                    * First name can not be empty
                  </p>
                ) : (
                  ''
                )}

                <input
                  type="text"
                  // onChange={(e) => setlastName(e.target.value)}
                  className=" w-full p-1 bg-background rounded-lg "
                  placeholder=" Last name "
                  {...register('lastName', {
                    required: true,
                  })}
                />
                {errors.lastName ? (
                  <p className="text-red-800 font-semibold text-[12px]">
                    * Last name can not be empty
                  </p>
                ) : (
                  ''
                )}
                <input
                  type="text"
                  // onChange={(e) => setcountry(e.target.value)}
                  className=" w-full p-1 bg-background rounded-lg"
                  placeholder=" Country "
                  {...register('country', {
                    required: true,
                  })}
                />
                {errors.country ? (
                  <p className="text-red-800 font-semibold text-[12px]">
                    * Country can not be empty
                  </p>
                ) : (
                  ''
                )}
                <input
                  type="text"
                  // onChange={(e) => setcity(e.target.value)}
                  className=" w-full p-1 bg-background rounded-lg"
                  placeholder=" City "
                  {...register('city', { required: true })}
                />
                {errors.city ? (
                  <p className="text-red-800 font-semibold text-[12px]">
                    * City can not be empty
                  </p>
                ) : (
                  ''
                )}
              </div>
            </div>
            {/* </div> */}

            <div className="flex items-center justify-end p-6 border-t border-solid border-slate-200 rounded-b">
              <button
                type="button"
                onClick={toggleForm}
                className="text-red-500 background-transparent font-bold uppercase px-6 py-2 text-sm outline-none focus:outline-none mr-1 mb-1 ease-linear transition-all duration-150"
              >
                Cancel
              </button>

              <button
                type="submit"
                // onClick={submitForm}
                className="bg-primary text-background active:bg-emerald-600 font-bold uppercase text-sm px-6 py-3 rounded shadow hover:shadow-lg outline-none hover:bg-tertiary hover:text-primary focus:outline-none mr-1 mb-1 ease-linear transition-all duration-150"
              >
                Save
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  );
}

export default SettingForm;
