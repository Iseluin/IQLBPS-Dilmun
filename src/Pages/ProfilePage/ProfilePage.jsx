import React from "react";
import Profile from "../../components/Profile/Profile";


function ProfilePage (){
    return(
        <div data-testid = 'Profile-page'>
            <Profile />
        </div>
    )} 
export default ProfilePage;