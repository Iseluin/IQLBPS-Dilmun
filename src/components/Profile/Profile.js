import React, { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import _ from 'lodash';
import { user } from '../../features/slices/user';
import { auth } from '../../firebase-config';
import { ADD_ITEM_ROUTE } from '../../route';
import defaultProfileImg from '../../assets/img/defaultProfileImg.jpg';
import defaultBGImg from '../../assets/img/defaultBGImg.jpg';
import CustomItemCard from '../CustomComponents/CustomItemCard';
import SettingForm from './SettingForm';

function Profile() {
  const userData = useSelector(user);
  const [email, setEmail] = useState();
  const [isOpen, setIsOpen] = useState(false);

  useEffect(() => {
    if (!_.isEmpty(userData)) {
      setEmail(auth.currentUser.email);
    }
  }, [email]);

  const toggleForm = () => {
    setIsOpen(!isOpen);
  };

  return (
    <div
      data-testid="profile"
      className=" bg-background bg-signin-background bg-cover bg-no-repeat w-full min-h-[100vh] flex flex-col justify-center items-center content-center "
    >
      <div className=" mt-5 mb-14 w-5/6 bg-background rounded-lg h-full relative">
        <div className="flex flex-col justify-center items-center rounded-lg ">
          <img
            src={defaultBGImg}
            alt="Background Img"
            className="bg-cover bg-center w-full min-h-[20vh] md:min-h-[40vh] rounded-lg "
          />
          <img
            src={defaultProfileImg}
            alt="Profile Img"
            className="rounded-full w-36 absolute flex outline outline-background  min-h-[15vh] md:min-h-[30vh] min-w-[15vh] md:min-w-[30vh]"
          />
        </div>
        <div className="flex flex-col items-center justify-center content-center  mb-14 bg-background rounded-lg  pt-4">
          <h1 className="font-bold text-[20px] sm:text-[26px] lg:text-[34px] text-primary m-1">
            {userData.firstName} {userData.lastName}
          </h1>
          <h3 className="font-semibold text-[14px] sm:text-[18px] lg:text-[24px] text-primary m-1">
            {userData.country ? userData.country : 'Country'}, City
          </h3>
          {console.log(userData.country)}
          <h5 className="font-semibold text-[14px] sm:text-[18px] lg:text-[24px] text-primary flex flex-row m-1">
            {' '}
            {email}
          </h5>
          <button
            type="button"
            onClick={toggleForm}
            className="text-primary hover:text-secondary"
          >
            <svg
              xmlns="http://www.w3.org/2000/svg"
              fill="none"
              viewBox="0 0 24 24"
              strokeWidth={1.5}
              stroke="currentColor"
              className="w-6 h-6 m-3"
            >
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                d="M10.343 3.94c.09-.542.56-.94 1.11-.94h1.093c.55 0 1.02.398 1.11.94l.149.894c.07.424.384.764.78.93.398.164.855.142 1.205-.108l.737-.527a1.125 1.125 0 011.45.12l.773.774c.39.389.44 1.002.12 1.45l-.527.737c-.25.35-.272.806-.107 1.204.165.397.505.71.93.78l.893.15c.543.09.94.56.94 1.109v1.094c0 .55-.397 1.02-.94 1.11l-.893.149c-.425.07-.765.383-.93.78-.165.398-.143.854.107 1.204l.527.738c.32.447.269 1.06-.12 1.45l-.774.773a1.125 1.125 0 01-1.449.12l-.738-.527c-.35-.25-.806-.272-1.203-.107-.397.165-.71.505-.781.929l-.149.894c-.09.542-.56.94-1.11.94h-1.094c-.55 0-1.019-.398-1.11-.94l-.148-.894c-.071-.424-.384-.764-.781-.93-.398-.164-.854-.142-1.204.108l-.738.527c-.447.32-1.06.269-1.45-.12l-.773-.774a1.125 1.125 0 01-.12-1.45l.527-.737c.25-.35.273-.806.108-1.204-.165-.397-.505-.71-.93-.78l-.894-.15c-.542-.09-.94-.56-.94-1.109v-1.094c0-.55.398-1.02.94-1.11l.894-.149c.424-.07.765-.383.93-.78.165-.398.143-.854-.107-1.204l-.527-.738a1.125 1.125 0 01.12-1.45l.773-.773a1.125 1.125 0 011.45-.12l.737.527c.35.25.807.272 1.204.107.397-.165.71-.505.78-.929l.15-.894z"
              />
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
              />
            </svg>
          </button>
          {isOpen && <SettingForm toggleForm={toggleForm} />}

          <div className=" items-center divide-x border-t border-primary ">
            <div className="flex-grow border-t border-primary divide-x" />

            <h3 className="flex-shrink mx-16 font-bold text-[14px] sm:text-[18px] lg:text-[24px] text-primary mt-3 mb-3">
              My Items
            </h3>
          </div>
          <div className="grid grid-cols-2 gap-4 md:grid-cols-3 p-5">
            <CustomItemCard />
            <CustomItemCard />
            <CustomItemCard />
            <CustomItemCard />
            <CustomItemCard />
          </div>
          <Link to={ADD_ITEM_ROUTE}>
            {' '}
            <button
              type="button"
              className="sm:w-96 w-80 font-semibold shadow-lg my-2 bg-primary text-background py-2 rounded-md hover:bg-tertiary hover:text-primary duration-500"
            >
              {' '}
              Add New Item{' '}
            </button>
          </Link>
        </div>
      </div>
    </div>
  );
}

export default Profile;
