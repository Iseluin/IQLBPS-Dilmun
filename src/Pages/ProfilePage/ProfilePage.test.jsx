import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import {store} from '../../app/store'
import ProfilePage from './ProfilePage';
import Profile from '../../components/Profile/Profile';

const MOCK_USER_ITEM = {
  title: 'Title',
  file: 'https://cdn.discordapp.com/attachments/1031834305703460906/1041710013992947812/image.png',
  description: 'Description',
  price: 110,
  type: 'Item Type',
  categories: ['Men'],
  owner: {
    email: 'email',
  },
};

const MOCK_USER_DATA = {
  firstName: 'user',
  lastName: 'user',
  address: {
    country:'user', city:'user'
  },
  email:'user@user.user'
}

test('Profile Page Rendered Correctly', () => {
    render(
      <Provider store={store}>
        <Router>
          <ProfilePage>
            <Profile userItem={MOCK_USER_ITEM} userData={MOCK_USER_DATA } />
          </ProfilePage>
        </Router>
      </Provider>
    );
    expect(screen.getByTestId('profile-page')).toMatchSnapshot();
  });
  