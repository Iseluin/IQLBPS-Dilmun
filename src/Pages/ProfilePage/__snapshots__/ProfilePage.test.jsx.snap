// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Page Rendered Correctly 1`] = `
<div
  data-testid="profile-page"
>
  <div
    class=" bg-background  bg-cover bg-no-repeat w-full h-full flex flex-col justify-center items-center content-center "
    data-testid="profile"
  >
    <div
      class=" mt-5 mb-14 w-5/6 bg-background rounded-lg h-full relative"
    >
      <div
        class="flex flex-col justify-center items-center rounded-lg "
      >
        <img
          alt="Background Img"
          class="bg-cover bg-center w-full min-h-[20vh] md:min-h-[40vh] rounded-lg "
          src="defaultBGImg.jpg"
        />
        <img
          alt="Profile Img"
          class="rounded-full w-36 absolute flex outline outline-background  min-h-[15vh] md:min-h-[30vh] min-w-[15vh] md:min-w-[30vh]"
          src="defaultProfileImg.jpg"
        />
      </div>
      <div
        class="flex flex-col items-center justify-center content-center  mb-14  rounded-lg  pt-4"
      >
        <h1
          class="font-bold text-[20px] sm:text-[26px] lg:text-[34px] text-primary m-1"
        >
           
        </h1>
        <button
          class="text-primary hover:text-secondary"
          type="button"
        >
          <svg
            class="w-6 h-6 m-3"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.343 3.94c.09-.542.56-.94 1.11-.94h1.093c.55 0 1.02.398 1.11.94l.149.894c.07.424.384.764.78.93.398.164.855.142 1.205-.108l.737-.527a1.125 1.125 0 011.45.12l.773.774c.39.389.44 1.002.12 1.45l-.527.737c-.25.35-.272.806-.107 1.204.165.397.505.71.93.78l.893.15c.543.09.94.56.94 1.109v1.094c0 .55-.397 1.02-.94 1.11l-.893.149c-.425.07-.765.383-.93.78-.165.398-.143.854.107 1.204l.527.738c.32.447.269 1.06-.12 1.45l-.774.773a1.125 1.125 0 01-1.449.12l-.738-.527c-.35-.25-.806-.272-1.203-.107-.397.165-.71.505-.781.929l-.149.894c-.09.542-.56.94-1.11.94h-1.094c-.55 0-1.019-.398-1.11-.94l-.148-.894c-.071-.424-.384-.764-.781-.93-.398-.164-.854-.142-1.204.108l-.738.527c-.447.32-1.06.269-1.45-.12l-.773-.774a1.125 1.125 0 01-.12-1.45l.527-.737c.25-.35.273-.806.108-1.204-.165-.397-.505-.71-.93-.78l-.894-.15c-.542-.09-.94-.56-.94-1.109v-1.094c0-.55.398-1.02.94-1.11l.894-.149c.424-.07.765-.383.93-.78.165-.398.143-.854-.107-1.204l-.527-.738a1.125 1.125 0 01.12-1.45l.773-.773a1.125 1.125 0 011.45-.12l.737.527c.35.25.807.272 1.204.107.397-.165.71-.505.78-.929l.15-.894z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
            <path
              d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
        <div
          class=" items-center divide-x border-t border-primary "
        >
          <div
            class="flex-grow border-t border-primary divide-x"
          />
          <h3
            class="flex-shrink mx-16 font-bold text-[14px] sm:text-[18px] lg:text-[24px] text-primary mt-3 mb-3"
          >
            My Items
          </h3>
        </div>
        <div
          class="grid xl:grid-cols-2 gap-4 p-5 max-w-screen-2xl	"
        />
        <a
          class="w-full text-center"
          href="/add-item"
        >
          <button
            class="w-1/2 p-4 font-semibold shadow-lg bg-primary text-background  rounded-md hover:bg-tertiary hover:text-primary duration-500"
            type="button"
          >
            Add New Item
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
